fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PartitionPruneRule.java

```diff
@@ -84,8 +97,70 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
+                                          LogicalOlapScanOperator olapScanOperator) {
+
+        Map<ColumnRefOperator, TreeMap<LiteralExpr, Set<Long>>> columnToPartitionValuesMap = new HashMap<>();
+        Map<ColumnRefOperator, Set<Long>> columnToNullPartitions = new HashMap<>();
+
+        // single item list partition has only one column mapper
+        Map<Long, List<LiteralExpr>> literalExprValuesMap = listPartitionInfo.getLiteralExprValues();
+        if (literalExprValuesMap != null && literalExprValuesMap.size() > 0) {
+            TreeMap<LiteralExpr, Set<Long>> partitionValueToIds = new TreeMap<>();
+            literalExprValuesMap.forEach((partitionId, values) ->
+                    values.forEach(value ->
+                            putValueMapItem(partitionValueToIds, partitionId, value)));
+            // single item list partition has only one column
+            Column column = listPartitionInfo.getPartitionColumns().get(0);
+            ColumnRefOperator columnRefOperator = olapScanOperator.getColumnReference(column);
+            columnToPartitionValuesMap.put(columnRefOperator, partitionValueToIds);
+            columnToNullPartitions.put(columnRefOperator, new HashSet<>());
+        }
+
+        // multiItem list partition mapper
+        Map<Long, List<List<LiteralExpr>>> multiLiteralExprValues = listPartitionInfo.getMultiLiteralExprValues();
+        if (multiLiteralExprValues != null && multiLiteralExprValues.size() > 0) {
+            List<Column> columnList = listPartitionInfo.getPartitionColumns();
+            for (int i = 0; i < columnList.size(); i++) {
+                TreeMap<LiteralExpr, Set<Long>> partitionValueToIds = new TreeMap<>();
+                for (Map.Entry<Long, List<List<LiteralExpr>>> entry : multiLiteralExprValues.entrySet()) {
+                    Long partitionId = entry.getKey();
+                    List<List<LiteralExpr>> multiValues = entry.getValue();
+                    for (List<LiteralExpr> values : multiValues) {
+                        LiteralExpr value = values.get(i);
+                        putValueMapItem(partitionValueToIds, partitionId, value);
+                    }
+                }
+                Column column = columnList.get(i);
+                ColumnRefOperator columnRefOperator = olapScanOperator.getColumnReference(column);
+                columnToPartitionValuesMap.put(columnRefOperator, partitionValueToIds);
+                columnToNullPartitions.put(columnRefOperator, new HashSet<>());
+            }
+        }
+
+        List<ScalarOperator> scalarOperatorList = Utils.extractConjuncts(olapScanOperator.getPredicate());
+        PartitionPruner partitionPruner = new ListPartitionPruner(columnToPartitionValuesMap,
+                columnToNullPartitions, scalarOperatorList);
+        try {
+            return partitionPruner.prune();
+        } catch (AnalysisException e) {
+            LOG.warn("PartitionPrune Failed. ", e);
+        }
+        return null;
```

Comment:
better way is return all partition?

Reply:
the rangePartitionPrune has return null partitions when throw exception. better to keep the same with the rang partition prune ？